### PR TITLE
don't allocate handle pids for external files

### PR DESCRIPTION
**Pull request by emanueldima**
solves https://github.com/EUDAT-B2SHARE/b2share/issues/1658

### CODE REVIEW

**Comment by emanueldima** [comment]
@nharraud Tests added

File: tests/b2share_unit_tests/records/test_records_api.py [hunk 1]
```diff
@@ -202,3 +202,40 @@ def test_record_delete_version(app, test_records, test_users):
...
+            deposit.publish()
+            deposit.commit()
+
+            _, record = deposit.fetch_published()
+
+            # external files don't get a handle PID, they already have one
+            # which is stored in record['_deposit']['external_pids']
+            for f in record.files:
+                if f['key'] in external_keys:
+                    assert f.get('ePIC_PID') is None
```
Comment (nharraud) [inline tests/b2share_unit_tests/records/test_records_api.py]: Won't the UI be missing the ePIC_PID in `_files`? We could also just copy the PID from external_pids.

Comment (emanueldima) [reply tests/b2share_unit_tests/records/test_records_api.py]: That is true. Right now we don't show any pids for external files. The tests would have to be a bit different then.

Comment (nharraud) [reply tests/b2share_unit_tests/records/test_records_api.py]: We can just change the PR by copying the PID from external_pids, which simplifies the work of 3rd party scripts. They will always be able to retrieve the ePIC_PID in `_files` that way.

**Comment by emanueldima** [comment]
I changed the serialization function to add the ePIC_PID of external files into the files section. With this change the UI works as expected.

File: tests/b2share_unit_tests/records/test_records_rest.py [hunk 2]
```diff
@@ -520,3 +520,46 @@ def test_access(user=None, deleted=True):
...
+
+            for f in record['files']:
+                assert f['ePIC_PID']
+                if f['key'] in uploaded_files:
+                    # uploaded (internal) file
+                    assert '0000' in f['ePIC_PID'] # freshly allocated fake pid
+                else:
+                    # external file
+                    assert f['b2safe']
+                    assert '0000' not in f['ePIC_PID'] # external pid
```
Comment (nharraud) [inline tests/b2share_unit_tests/records/test_records_rest.py]: Why not testing directly if the PID is the same?
```
assert f['ePIC_PID'] == [rec for rec in records[external_pids] if rec['key'] == f['key']][0]['ePIC_PID']
```

Comment (emanueldima) [reply tests/b2share_unit_tests/records/test_records_rest.py]: yep, fixed

File: tests/b2share_unit_tests/records/test_records_rest.py [hunk 3]
```diff
@@ -520,3 +520,48 @@ def test_access(user=None, deleted=True):
...
+                assert f['ePIC_PID']
+                if f['key'] in uploaded_files:
+                    # uploaded (internal) file
+                    assert '0000' in f['ePIC_PID'] # freshly allocated fake pid
+                else:
+                    # external file
+                    assert f['b2safe']
+                    external_pid = [rec for rec in external_pids
+                                    if rec['key'] == f['key']]
+                    assert f['ePIC_PID'] == external_pid[0]['ePIC_PID']
```
Comment (nharraud) [inline tests/b2share_unit_tests/records/test_records_rest.py]: There are two different external PIDs in this record: https://github.com/EUDAT-B2SHARE/b2share/blob/master/tests/conftest.py#L352-L359

How is it possible that every file in record['files'] has the same external PID value?

Comment (emanueldima) [reply tests/b2share_unit_tests/records/test_records_rest.py]: `external_pid` is the one that matches the current file (defined just above). It's distinct from `external_pids` (it's not a great name, I admit).

Comment (emanueldima) [reply tests/b2share_unit_tests/records/test_records_rest.py]: Actually it's better to just change this name.

**Comment by nharraud** [comment]
Thanks for the changes :)

**Pull request merged**